crates/sui-framework/sources/tx_context.move

```diff
@@ -41,6 +43,13 @@ module sui::tx_context {
         self.sender
     }
 
+    /// Return the recipient for the user that signed the current
+    /// transaction
+    public fun recipient(self: &TxContext): Recipient {
+        sui::address::recipient(self.sender)
```

Comment:
Are there cases (in the future) where this is not the sender address, or it's actually an object recipient?

Reply:
I think if we ever want to build something like account abstraction, we will need ability to set a "sender" that is not an address. 
But I'm not exactly sure if in such a world sender would even make sense? 
We could just make `sender` return a `Recipient` if we wanted to be safe. 

This to me is the huge "unknown" area of this PR, and it's all very speculative